core/services/ocr2/plugins/directrequestocr/test/drocr_integration_test.go

```diff
@@ -49,17 +48,23 @@ func TestIntegration_OCR2DR_MultipleRequests_Success(t *testing.T) {
 		jobIds = append(jobIds, ocrJob.ID)
 	}
 
+	// config for registry contract
+	utils.SetRegistryConfig(t, owner, registryContract)
+
 	// config for oracle contract
-	utils.SetConfig(t, owner, oracleContract, oracles)
+	utils.SetOracleConfig(t, owner, oracleContract, oracles)
 	utils.CommitWithFinality(b)
 
+	// set up subscription
+	subscriptionId := utils.CreateAndFundSubscriptions(t, owner, linkToken, registryContract)
+
 	// send requests
 	sent := make([][]byte, nClients)
 	s := rand.NewSource(666)
 	r := rand.New(s)
 	for i := 0; i < nClients; i++ {
 		sent[i] = []byte{byte(r.Uint32() % 256)}
-		_, err := clientContracts[i].Contract.SendRequest(owner, hex.EncodeToString(sent[i]), []byte{}, []string{}, big.NewInt(3))
+		_, err := clientContracts[i].Contract.SendRequest(owner, hex.EncodeToString(sent[i]), []byte{}, []string{}, subscriptionId)
```

Comment:
A local test run breaks here with "execution reverted".

Reply:
Two things I discovered so far:

1.  You're not calling setAuthorizedSenders 
`registryContract.SetAuthorizedSenders(owner, []common.Address{ocrContractAddress})`

2. You're not calling addConsumer but I don't know how to fix it. Why doesn't the following work?
`registryContract.AddConsumer(owner, 1, clientContractAddress)`

Reply:
> You're not calling addConsumer but I don't know how to fix it. Why doesn't the following work?
Might be because `1` needs to be cast to uint64

Reply:
The problem was that I was trying to call AddConsumer before CreateSubscription.

After your latest fixes I'm hitting InsufficientBalance.

Reply:
Oh woah, there are 20 clients. Added more funding based on that number.

Reply:
That doesn't solve it, there must be some other problem.